src/files/glob-source.js

```diff
@@ -77,15 +107,29 @@ async function * toGlobSource ({ path, type, prefix }, options) {
 
   const globOptions = Object.assign({}, options.glob, {
     cwd: path,
-    nodir: true,
+    nodir: false,
     realpath: false,
     absolute: true
   })
 
   for await (const p of glob(path, '**/*', globOptions)) {
+    if (preserveMode || preserveMtime) {
+      const stat = await fs.stat(p)
+
+      if (options.preserveMode) {
+        mode = stat.mode
+      }
+
+      if (options.preserveMtime) {
+        mtime = parseInt(stat.mtimeMs / 1000)
+      }
+    }
+
     yield {
       path: toPosix(p.replace(prefix, '')),
-      content: fs.createReadStream(p)
+      content: fs.createReadStream(p),
```

Comment:
Now that `nodir: false`, will this try to read a directory?

Reply:
Yes, although not now cos it was an bug and has been squashed with tests added.